File: app/src/main/java/nl/jpelgrm/movienotifier/data/APIHelper.java
package nl.jpelgrm.movienotifier.data;

import com.google.gson.Gson;
import com.google.gson.GsonBuilder;

import nl.jpelgrm.movienotifier.BuildConfig;
import nl.jpelgrm.movienotifier.util.StethoUtil;
import retrofit2.Retrofit;
import retrofit2.converter.gson.GsonConverterFactory;

public class APIHelper {
    private static APIClient instance = null;

    public static APIClient getInstance() {
        if(instance == null) {
            Gson gson = new GsonBuilder().serializeNulls().excludeFieldsWithoutExposeAnnotation().create();

            Retrofit retrofit = new Retrofit.Builder()
                    .baseUrl(BuildConfig.SERVER_BASE_URL)
                    .client(StethoUtil.getOkHttpClient())
                    .addConverterFactory(GsonConverterFactory.create(gson))
                    .build();

            instance = retrofit.create(APIClient.class);
        }
        return instance;
    }
}

File: app/src/main/java/nl/jpelgrm/movienotifier/data/APIHelper.kt
package nl.jpelgrm.movienotifier.data

import nl.jpelgrm.movienotifier.BuildConfig
import retrofit2.Retrofit
import retrofit2.converter.moshi.MoshiConverterFactory

object APIHelper {
    val instance: APIClient by lazy {
        val retrofit = Retrofit.Builder()
                .baseUrl(BuildConfig.SERVER_BASE_URL)
                .addConverterFactory(MoshiConverterFactory.create())
                .build()

        retrofit.create(APIClient::class.java)
    }
}